### PR TITLE
New eslint rules

### PR DESCRIPTION
**Pull request by sampaiodiego**
I'm proposing adding this bunch of new eslint rules to better standardize our code base and also making PR reviews a bit easier (since we don't need to care about much of code styling).

_Since this is still a proposition, I'm not worried of fixing the issues. I have fixed most of them using `--fix`, but some need manual fix which I'm willing to do since we all approve the new rules._

For easy reviewing, I'm proposing adding the following rules:

* [no-duplicate-imports](https://eslint.org/docs/rules/no-duplicate-imports)
* [no-multiple-empty-lines](https://eslint.org/docs/rules/no-multiple-empty-lines) `{ "max": 2 }`
* [arrow-parens](https://eslint.org/docs/rules/arrow-parens) `"always"`
* [quote-props](https://eslint.org/docs/rules/quote-props) `"as-needed"`
* [no-array-constructor](https://eslint.org/docs/rules/no-array-constructor)
* [array-callback-return](https://eslint.org/docs/rules/array-callback-return)
* [prefer-destructuring](https://eslint.org/docs/rules/prefer-destructuring) `{"object": true, "array": false}`
* [prefer-rest-params](https://eslint.org/docs/rules/prefer-rest-params)
* [arrow-spacing](https://eslint.org/docs/rules/arrow-spacing)
* [arrow-body-style](https://eslint.org/docs/rules/arrow-body-style) `"as-needed"`
* [no-confusing-arrow](https://eslint.org/docs/rules/no-confusing-arrow)
* [dot-notation](https://eslint.org/docs/rules/dot-notation)
* [no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary)
* [no-unneeded-ternary](https://eslint.org/docs/rules/no-unneeded-ternary)
* [spaced-comment](https://eslint.org/docs/rules/spaced-comment)
* [space-infix-ops](https://eslint.org/docs/rules/space-infix-ops)
* [array-bracket-spacing](https://eslint.org/docs/rules/array-bracket-spacing) `"never"`
* [object-curly-spacing](https://eslint.org/docs/rules/object-curly-spacing) `"always"`
* [comma-dangle](https://eslint.org/docs/rules/comma-dangle) **changed** from `"never"` to `"always-multiline"`
* [no-mixed-operators](https://eslint.org/docs/rules/no-mixed-operators) added based on [airbnb](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/style.js#L287)

@RocketChat/core please evaluate.

### CODE REVIEW

**Comment by sampaiodiego** [comment]
guys, I've added a new rule [no-mixed-operators](https://eslint.org/docs/rules/no-mixed-operators)  based on [airbnb](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/style.js#L287)

the pull request is ready to be reviewed.. I recommend looking at particularly commits, since most of the fixes were by `--fix`, so here is a short list of what I've changed manually:
* general issues - 7d3c44730b23f5f9ceb6a1df12fba6fbd6caf61f
* destructuring - 844151f4d821d2bfb2e48ecff769b072c8fa6a8e
* `no-confusing-arrow` - 39c1b92b869bf701f5da6577c5bac2fd5c443395
* remaining issues - 50a6cf1c66abd8a3eb80aa342374886f88ecaf4b
* `no-mixed-operators` - c8993331e8144d1c69d1677dcca5d3f03a42d09e

File: packages/rocketchat-lib/server/functions/saveUser.js [hunk 1]
```diff
@@ -141,9 +141,9 @@ RocketChat.saveUser = function(userId, userData) {
 
 			subject = RocketChat.placeholders.replace(subject);
 			html = RocketChat.placeholders.replace(html, {
-				name: s.escapeHTML(userData.name),
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/functions/saveUser.js]: Is this change right?

File: packages/rocketchat-lib/server/functions/setUserAvatar.js [hunk 2]
```diff
@@ -35,8 +35,8 @@ RocketChat.setUserAvatar = function(user, dataURI, contentType, service) {
 	} else {
 		const fileData = RocketChatFile.dataURIParse(dataURI);
 		encoding = 'base64';
-		image = fileData.image;
-		contentType = fileData.contentType;
+		({ image } = fileData);
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/functions/setUserAvatar.js]: Do we need parenthesis in these cases?

**Comment by sampaiodiego** [comment]
closing this in favor of https://github.com/RocketChat/Rocket.Chat/pull/11804

**Pull request closed without merging**